### DISCUSSION TITLE
Changing Parameterization Schemes

### DISCUSSION THREAD

**Discussion by bnepal22** [ufs-community/ufs-weather-model#2393] [Q&A] [answered]
How to change the current parameterization scheme in the UFS model. Currently, I am using the FV3_GFS_v15p2 version. Is there any detail process manual for doing so?

**Comment by gspetro-NOAA** [comment]
Hi @bnepal22 ,

My understanding is that you can just indicate a different suite wherever `FV3_GFS_v15p2` appears in your previous configuration file(s). In general, the [build/run chapter of the User's Guide](https://ufs-weather-model.readthedocs.io/en/develop/BuildingAndRunning.html#setting-the-cmake-flags-and-ccpp-suites-environment-variables) has some information on how to set the `CMAKE_FLAGS` if you are compiling using `build.sh`. Specifically, you would change the `-DCCPP_SUITES` flag (e.g., `-DCCPP_SUITES=FV3_GFS_v16`). That chapter also has information on [running regression tests](https://ufs-weather-model.readthedocs.io/en/develop/BuildingAndRunning.html#running-the-model), which you could try if you are running an existing/supported configuration and just switching out the physics. In that case, you would add the physics suite you want to use in the COMPILE statement of your configuration file if it isn't already listed (e.g., `-DCCPP_SUITES=FV3_GFS_v16`, and you would add or modify the `CCPP_SUITE` variable in the test you are using (e.g., `CCPP_SUITE=FV3_GFS_v16`). 

Note that not all physics suites are appropriate for a given configuration, but without additional information (e.g., on machine, configuration, method for running), it would be difficult to advise you on which suites to use. In general, the [CCPP forum](https://github.com/NCAR/ccpp-physics/discussions/categories/q-a) is the best place to ask questions like that. 

@jkbk2004 Do you have anything to add here? 

Best,
Gillian

**Reply by jkbk2004** [reply]
@bnepal22 More information at CCPP documents: https://dtcenter.ucar.edu/GMTB/v6.0.0/sci_doc/index.html and https://ccpp-techdoc.readthedocs.io/en/v6.0.0/

**Reply by bnepal22** [reply]
I am sorry for not being clear in the previous question. I was trying to switch to Thompson Cloud Microphysics from GFDL Microphysics just for some test, and when I only changed the  "imp_physics = 11" to "imp_physics = 8" under "&gfs_phys_nml" in the namelist file, but it showed error. I'd like to know if there is any detailed manual.

**Reply by gspetro-NOAA** [reply]
@mkavulich Is this question best posted in the [CCPP-physics](https://github.com/NCAR/ccpp-physics/discussions/categories/q-a) discussions? Regardless, could you advise?
@bnepal22 I know that for the SRW App, there is some guidance [here](https://ufs-srweather-app.readthedocs.io/en/develop/UsersGuide/Reference/FAQ.html#advanced-how-can-i-add-a-physics-scheme-e-g-ysu-pbl-to-the-ufs-srw-app), but I think that if you want to change it in the WM itself, you would have to turn on (and off) certain physics-related options in your test file and probably update the CCPP suite definition file. In general, we encourage users to stick to the well-tested physics suites already in use, but we have definitely had users with reasons to change the usual combinations (e.g., for research purposes).

**Reply by bnepal22** [reply]
Yes, I am attempting such a physics change for research purposes, and basically, I am doing it for S2S scale prediction using a stretched grid.

**Reply by mkavulich** [reply, marked as answer]
@gspetro-NOAA I think this question is best answered here since it does have some dependence on the specific dycore.

@bnepal22 For models that use CCPP (as UFS weather model does), the physics suites (i.e. collections of schemes) that can be run are set at compile time with Suite Definition Files. So in order to use a brand new suite (which is typically the case if you are simply swapping out

In order to change just the microphysics scheme, unless there is an existing suite that uses the exact schemes you require, you will need to update the CCPP suite file (or create a new one) and re-compile. In this case, there is an existing suite `suite_FV3_GFS_v15_thompson_mynn_lam3km.xml` that uses Thompson, but it also has some other changes, so I would recommend creating a new file so that you can still use the old and new suites interchangeably.

From the ufs-weather-model directory:

```
cd FV3/ccpp/suites
cp suite_FV3_GFS_v15p2.xml suite_FV3_GFS_v15p2_thompson.xml
```

Then edit the new file to swap out GFDL microphysics with Thompson. The Thompson scheme actually consists of three schemes that must be run in order, so those changes will look like this:

```
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
-<suite name="FV3_GFS_v15p2" version="1">
+<suite name="FV3_GFS_v15p2_thompson" version="1">
   <!-- <init></init> -->
   <group name="fast_physics">
     <subcycle loop="1">
@@ -77,7 +77,9 @@
       <scheme>GFS_suite_interstitial_4</scheme>
       <scheme>cnvc90</scheme>
       <scheme>GFS_MP_generic_pre</scheme>
-      <scheme>gfdl_cloud_microphys</scheme>
+      <scheme>mp_thompson_pre</scheme>
+      <scheme>mp_thompson</scheme>
+      <scheme>mp_thompson_post</scheme>
       <scheme>GFS_MP_generic_post</scheme>
       <scheme>maximum_hourly_diagnostics</scheme>
       <scheme>GFS_physics_post</scheme>
```

Next, you will need to re-compile with this new suite. The easiest way to do this will be to set the CCPP_SUITES variable manually as was described above to include the new `FV3_GFS_v15p2_thompson` suite. This will depend on the context in which you are running the model: are you running the UFS weather model stand-alone or with one of the UFS applications?

**Reply by bnepal22** [reply]
I think it's stand-alone since I am running the Atmospheric model.
